packages/next-intl/src/middleware/middleware.tsx

```diff
@@ -168,9 +171,10 @@ export default function createMiddleware(config: MiddlewareConfig) {
         }
       } else {
         if (
-          hasMatchedDefaultLocale &&
-          (configWithDefaults.localePrefix === 'as-needed' ||
-            configWithDefaults.domains)
+          configWithDefaults.localePrefix === 'never' ||
+          (hasMatchedDefaultLocale &&
+            (configWithDefaults.localePrefix === 'as-needed' ||
+              configWithDefaults.domains))
```

Comment:
Alternate links don't really make sense in this case, since the URL won't set or change the current locale, right?

Reply:
Or you could actually. It's some 'hidden' functionality at the moment I believe. When you go to a URL prefixed with a locale it will set that locale in the cookie and redirect to the URL without the locale prefix.

Reply:
Maybe more explicit then, like this?

```
Link: 
<http://localhost:3000/de/about>; rel="alternate"; hreflang="de",
<http://localhost:3000/en/about>; rel="alternate"; hreflang="en",
<http://localhost:3000/about>; rel="alternate"; hreflang="x-default"
```

Reply:
Cannot find anything about screen readers and the use of alternate link headers though.

Reply:
Maybe leave it out for locale prefix `never`?

Reply:
Sure, that's the other option. Fine by me, but we should mention it in the docs then!

Reply:
Disabling the alternate links and a brief mention in the docs would be the last part missing here, afterwards we can release this!